bin/weewx/drivers/vantage.py

```diff
@@ -1110,14 +1212,27 @@ def hardware_name(self):
     def archive_interval(self):
         return self.archive_interval_
     
+    def determine_hardware(self):
```

Comment:
I wanted to repeat this code upto max_tries and because my lack on coding skills I didn't know how to do that otherwise.